docs/tasks.md

```diff
@@ -576,7 +576,7 @@ Param values from resources can also be accessed using [variable substitution](.
 
 ##### Variable Substitution with Parameters of Type `Array`
 
-Referenced parameters of type `array` will expand to insert the array elements in the reference string's spot.
+Referenced parameters of type `array` can be expanded using 'star-expansion' by adding `[*]` to the named parameter to insert the array elements in the reference string's spot.
```

Comment:
What happens if I don't use the * with an array param? Is that an error or will it simply not expand the values out?

Reply:
Yes with caveats. Right now the behaviour is identical as the star expansion is effectively an alias, but longer term when we add proper type support if you just reference the array directly then the array itself will be added. Whether or not that is an error is going to depend on the type we unmarshal into (see #1530 for where we're heading).

Reply:
As it turn out right now you can't assign an array parameter to an actual array field, so afaik in all cases how we are currently use arrays we should just use star-expansion. Talking to @afrittoli earlier, I'll also update the catalog tasks, but that is not desperately urgent for now other than to illustrate good practices.